src/app/src/main/java/br/com/italoromeiro/arctouchtest/presenters/BasePresenter.java

```diff
@@ -0,0 +1,74 @@
+package br.com.italoromeiro.arctouchtest.presenters;
+
+import android.util.Log;
+
+import org.androidannotations.annotations.Background;
+import org.androidannotations.annotations.Bean;
+import org.androidannotations.annotations.EBean;
+import org.androidannotations.annotations.RootContext;
+
+import br.com.italoromeiro.arctouchtest.presenters.actions.NetworkAction;
+import br.com.italoromeiro.arctouchtest.rest.RestController;
+import br.com.italoromeiro.arctouchtest.views.activities.BaseActivity;
+
+/**
+ * Created by italo on 08/08/16.
+ */
+@EBean
+public class BasePresenter {
```

Comment:
I like to use the MVP architecture. But I didn't use it formally in this example. Nevertheless, I decided to use "a part" of it. In this code the objective of BasePresenter is only take care of my network connections.

Formally, the MVP promises a gain of organization when you separate the concepts of "what is view interaction" and "what is business logic". A result of this, for instance, is the possibility of writing unit tests to test your logic instead of instrumentation tests that deal with it. But remember, I didn't do these tests in this exercise.

Another point to talk about is that I use this class to catch the exceptions that may occur and I do the treatment I want with them and, if appropriate, I can retry the last action I've done. Actually, the last part of that statement wasn't implemented yet; so, I think, you might be confused about the meaning of my "retriable".

Like I said in the next comment, when I catch an exception, I can do the treatment using the doError and, if appropriate, I can do a new try using that "retriable". I can do this because the treatment over my exception might repair a possible bad state of my application and, if all is good, I can try my action again and all of this without boring my client with a series of popups.

If I'm not clear enough, please, you may question me about more details. I like to discuss how can I do better my treatment of exceptions.